app/view_models/costs/month_item.rb

```diff
@@ -4,34 +4,27 @@ class MonthItem
 
     delegate :id, to: :@cost_item, prefix: 'cost_item'
 
-    def initialize(cost_item:, year_month:)
+    def initialize(cost_item:, year_month:, costs:)
       @cost_item = cost_item
       @year_month = year_month
+
+      @data = (costs || []).each_with_object(forecast: false, value: 0, count: 0) do |cost, data|
```

Comment:
Could you default `costs: []` so this could be simplified to `costs.each_with_object(...)`?